src/sentry/api/bases/user.py

```diff
@@ -26,7 +26,7 @@ def has_object_permission(self, request: Request, view, user: User | RpcUser | N
             return True
         if request.auth:
             return False
-        if is_active_superuser(request):
+        if superuser_has_permission(request):
             return True
         return False
```

Comment:
nit: If we could combine all these conditionals together, it would make it easier to read and also help give order of operation precedence! Also seems like we don't need/reference `self`, so could this be a `cls` classmethod, or do we think we'll need other references in the future?

Reply:
i think all other reimplementations of `has_object_permission` use `self`, so i'm not sure if that's a DRF requirement or convenetion here. i'm also not sure if the precedence is there for a reason though so i'm a bit hesitant to refactor

Reply:
in any case i kinda like how it checks for each condition explicitly